modules/c++/sys/source/OSUnix.cpp

```diff
@@ -132,13 +133,68 @@ std::set<std::string> get_unique_thread_siblings()
     return unique_ts;
 }
 
-void get_cpu_affinity(cpu_set_t* mask)
+class ScopedCPUAffinity
 {
-    if (-1 == sched_getaffinity(0, sizeof(cpu_set_t), mask))
+public:
+    ScopedCPUAffinity()
     {
-        throw except::Exception(Ctxt("Failed to get CPU affinity"));
+        int numOnlineCPUs = sysconf(_SC_NPROCESSORS_ONLN);
+        if (numOnlineCPUs == -1)
+        {
+            throw except::Exception(Ctxt("Failed to get online CPU count"));
+        }
+
+        // The number of processors in the mask must be _at least_ the number of
+        // CPU's representable by the kernel's internal mask. This is given by
+        // the constant CPU_SETSIZE.
+        int maxCPUs = std::max(numOnlineCPUs, CPU_SETSIZE);
+        mSize = CPU_ALLOC_SIZE(maxCPUs);
+        mMask = CPU_ALLOC(maxCPUs);
+
+        if (-1 == sched_getaffinity(0, mSize, mMask))
+        {
+            std::ostringstream msg;
+            msg << "Failed to get CPU affinity with"
+                << " CPU_SETSIZE=" << CPU_SETSIZE << ","
+                << " numOnlineCPUs=" << numOnlineCPUs << ","
+                << " alloc size=" << mSize << ".";
+            switch (errno)
+            {
+                case (EINVAL) :
```

Comment:
Don't indent the `case` inside `switch` statements (so move everything inside the body of the `switch` statement four spaces to the left)

Reply:
Don't put parens around the values (i.e. `case EINVAL:`)